### PR TITLE
Unify `TracePredicate`'s outputs

### LINKED ISSUE #55: Unify `TracePredicate`'s output
This issue is about code maintainability improvement at module `Plutus.Contract.Test.Extra`

#  What do we have:

Currently at any `*AtComputedAddress*` function body we see almost the same code:
```haskell
    unless result $
      tell @(Doc Void)
        ( -- here we creating some pretty message
        )
    return result
```
The cons of this is that:
- describing prettyprinting inside of a predicate, which is different responsibilities
- every function do that little bit different, so they are hard to generalize
#  What to do:
I propose to make a types `CheckedState` and `CheckedData` with `Pretty` instances, that would allow instead of writing
```haskell
"Data at address" <+> pretty addr <+> "was"
  <+> foldMap (foldMap pretty . Ledger.txData . Ledger.txOutTxTx) utxoMap
  <> line
  <> "Contract writer data was"
  <+> pretty w
```
write 
```haskell
pretty $ CheckedState addr [CheckedDatas utxoMap, CheckedWriter w]
```
This way it's more about describing what state to show, rather than how exactly to do it.

### PR DESCRIPTION
**Pull request by Renegatto**
Closes #55

### CODE REVIEW

File: plutus-extra/src/Plutus/Contract/Test/Extra.hs [hunk 1]
```diff
@@ -267,57 +294,30 @@ dataAtComputedAddressWithState ::
...
+          addr
+          (any (check w) datums)
 
 {- | Check that the UTxO at a computed address
  and data aquired from contract's writer instance meet some condition.
  The address is computed using data acquired from contract's writer instance.
 
   @since 1.1
 -}
+showStateIfFailAndReturn ::
```
Comment (kozross) [inline plutus-extra/src/Plutus/Contract/Test/Extra.hs:310]: Did the Haddock above become detached from the function it was supposed to document?

Comment (Renegatto) [reply plutus-extra/src/Plutus/Contract/Test/Extra.hs:310]: Oh right, it was :sweat_smile:
Gonna fix that.

**Comment by kozross** [comment]
@Renegatto please fix conflicts. Then I can merge.

**Comment by Renegatto** [comment]
Looks like I did break this branch, but conflicts resolved in #64

**Pull request closed without merging**